### ISSUE TITLE
Int instead of string added to a document

### ISSUE THREAD

**Issue by wino45** [TomWright/dasel#327]
**Describe the bug**
When I try to put "string" value while the string happens is a valid int for Go, the int is put in the document.
This is not expected, because, one cannot add a string (singe character) like this: '0'.

**To Reproduce**
```
echo 't:' | dasel put -r yaml -t string -v "0x12_11" -s 't' | dasel -r yaml 't.type()'
number
```

Since the string is provided, I have expected string on output, not int.
Same for bool:

```
echo 't:' | dasel put -r yaml -t string -v "true" -s 't' | dasel -r yaml 't.type()'
bool
```

As a bonus two panics:
`echo '' | dasel put -r yaml -t string -v "0x12_11" -s 't'`
`panic: reflect: call of reflect.Value.Set on zero Value`

`echo 't:' | dasel put -r yaml -t string -v "0x12_11" -s ''`
`panic: reflect: reflect.Value.Set using unaddressable value`

**Expected behavior**
Add string to a document as requested by user. 
Add value to an empty document on first panic.
Do not panic on second panic.

**Desktop (please complete the following information):**

- Windows 11
- dasel version v2.3.3

**Comment by TomWright** [comment]
Thanks for this.

The YAML parser used really dictates whether something is a number or string, but I can take a look to see if I can work around this issue.

Re the panics, potentially related to https://github.com/TomWright/dasel/issues/282

**Comment by TomWright** [comment]
The number is is resolved in v2.3.5. I missed the bool issue on the first pass and will resolve it shortly.

**Comment by TomWright** [comment]
This has been released under `v.2.4.0`.
